plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -336,7 +341,14 @@ public void commentColumn(String databaseName, String tableName, String columnNa
         io.trino.hive.thrift.metastore.Table table = delegate.getTable(databaseName, tableName)
                 .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
 
-        for (FieldSchema fieldSchema : table.getSd().getCols()) {
+        List<FieldSchema> fieldSchemas = ImmutableList.<FieldSchema>builder()
+                .addAll(table.getPartitionKeys())
+                .addAll(table.getSd().getCols())
+                .build();
+
+        verifyDistinctNames(fieldSchemas);
```

Comment:
@ebyhr does it make sense for Thrift?

Reply:
Tried at the HMS database level to perform the change of having two namesake columns of a table without succcess:

```
MariaDB [metastore]> select * from COLUMNS_V2 where CD_ID = 56;
+-------+---------+-------------+-----------+-------------+
| CD_ID | COMMENT | COLUMN_NAME | TYPE_NAME | INTEGER_IDX |
+-------+---------+-------------+-----------+-------------+
|    56 | NULL    | c1          | int       |           0 |
|    56 | NULL    | c2          | int       |           1 |
|    56 | NULL    | c3          | int       |           2 |
+-------+---------+-------------+-----------+-------------+
3 rows in set (0.00 sec)

MariaDB [metastore]> update COLUMNS_V2 set COLUMN_NAME = 'c2' where COLUMN_NAME = 'c1' and CD_ID = '56';
ERROR 1062 (23000): Duplicate entry '56-c2' for key 'PRIMARY'
MariaDB [metastore]> show create table COLUMNS_V2;
+------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table      | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
+------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| COLUMNS_V2 | CREATE TABLE `COLUMNS_V2` (
  `CD_ID` bigint(20) NOT NULL,
  `COMMENT` varchar(256) CHARACTER SET latin1 COLLATE latin1_bin DEFAULT NULL,
  `COLUMN_NAME` varchar(767) CHARACTER SET latin1 COLLATE latin1_bin NOT NULL,
  `TYPE_NAME` mediumtext,
  `INTEGER_IDX` int(11) NOT NULL,
  PRIMARY KEY (`CD_ID`,`COLUMN_NAME`),
  KEY `COLUMNS_V2_N49` (`CD_ID`),
  CONSTRAINT `COLUMNS_V2_FK1` FOREIGN KEY (`CD_ID`) REFERENCES `CDS` (`CD_ID`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |

```

I'd say that also for Thrift the check is not necessary.

Reply:
@findinpath thank you for checking that! 
I'm having my hands on removing that check, but I want to check one concern before: 
Do you think we can assume that constraint on column names will be created by all supported thrift metastores trino works with? e.g. different distributions?